### ISSUE TITLE
[Bug] "Object is possibly 'undefined'" in hint-compat

### ISSUE THREAD

**Issue by molant** [webhintio/hint#1661]
When I try to build `hint-compat-api` I get a `"Object is possibly 'undefined'"` error in [`getProblemLocationFromNode`](https://github.com/webhintio/hint/blob/26310655df3a20fcf95ddc2936fcb45f68c2a6da/packages/hint-compat-api/src/helpers/compat-css.ts#L34)

```ts
private getProblemLocationFromNode(node: ChildNode): ProblemLocation | undefined {
        const start = node.source.start;

        if (!start) {
            return undefined;
        }

        return {
            column: start.column - 1,
            line: start.line - 1
        };
    }
```

It happens in 2 separate machines. @borgitas21, @antross can you please confirm and do a PR with a fix (don't know if just adding `!`) is enough in this situation.

Thank you!

**Comment by antross** [comment]
At first glance this looks like an issue due to a change in type definitions in a patch update of `postcss`. It just bumped to `7.0.8` two days ago (we reference `7.0.7`).

I should have a PR ready shortly.

**Comment by antross** [comment]
For reference, here's the PR that broke us (released as part of `postcss` `7.0.8`):
https://github.com/postcss/postcss/pull/1217

It corrects the type definitions to match what `postcss` was already doing in previous versions (they were out-of-sync).
